From stable-bounces@linux.kernel.org Wed Feb  6 21:43:41 2008
From: Dave Airlie <airlied@linux.ie>
Date: Thu, 7 Feb 2008 05:43:11 +0000 (GMT)
Subject: [stable] drm: the drm really should call pci_set_master..
To: stable@kernel.org
Message-ID: <alpine.DEB.0.82.0802070541260.26136@skynet.skynet.ie>

From: Dave Airlie <airlied@linux.ie>


(submitted upstream as 19a8f59ab8ceee751ea720085098355d53f727d6)

perhaps bonghits could turn on my bus-mastering because the drm
certainly never bothered doing it before.

Signed-off-by: Dave Airlie <airlied@linux.ie>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/char/drm/drm_stub.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/char/drm/drm_stub.c
+++ b/drivers/char/drm/drm_stub.c
@@ -218,6 +218,7 @@ int drm_get_dev(struct pci_dev *pdev, co
 	if (ret)
 		goto err_g1;
 
+	pci_set_master(pdev);
 	if ((ret = drm_fill_in_dev(dev, pdev, ent, driver))) {
 		printk(KERN_ERR "DRM: Fill_in_dev failed.\n");
 		goto err_g2;
